### DISCUSSION TITLE
Dexcom Transmitter ID Detected as a Password

### DISCUSSION THREAD

**Discussion by databoy2k** [NightscoutFoundation/xDrip#3488] [Bug reports]
This is a super low priority issue, but I'm wondering if the field for the Dexcom transmitter ID is for some reason being detected as a password.

Every time I enter or change the transmitter ID, my password manager, Enpass, prompts me to save the new transmitter ID into it. See the screenshot below: 
![Screenshot_20240526-203154](https://github.com/NightscoutFoundation/xDrip/assets/18686442/7953adbf-b18f-431a-92dd-978298b4903f)

Clicking save does nothing. It's obviously not getting the right details, but something is triggering the password manager to think that the transmitter ID field is a password of some sort. 

Anyone else getting this?

**Comment by databoy2k** [comment]
Could someone maybe point me to the right area of the code where transmitter IDs are entered? I could try to follow the routine and see what is tripping Enpass.

I assume other password managers (bitwarden, etc.) are not affected. Right?

**Reply by Navid200** [reply]
Are you sure this is an xDrip issue and not a phone issue?

I have used xDrip entering Dexcom serial numbers for years.
I have never experienced this.

I have done this from Android 5 to currently 15 beta.
And I have done it with Motorola, Asus and Pixel.

If you can address this on your phone, it would be the preferred solution.

**Reply by databoy2k** [reply]
It might be an issue with that particular password manager. I don't use any of the other ones like bitwarden or 1password. I asked because if other apps that support similar functionality do not respond to the field, then it must just be an Enpass issue.

Sadly it's not something that can be changed on a per device basis. I thought there must be some attribute specific to that field that Enpass picks up as a password.

**Reply by Navid200** [reply]
Why aren't we hearing this from anyone else?

If indeed something is wrong with xDrip, we will need a demonstration or a description of how to recreate the problem.

Unless you are using a phone that no one else has access to, or an app that no one else uses.

Why do you need a password manager?
Why can it not understand that the filed is not a password if you tell it once?
Have you done a search to see if others have similar issues with your password manager?

**Reply by Navid200** [reply]
This is where the transmitter ID is entered.
https://github.com/NightscoutFoundation/xDrip/blob/6e1f0651bc8e765b3fe58a038e2ec2084d57e2bc/app/src/main/res/xml/pref_data_source.xml#L165-L172

And indeed it has password as one of the input types.

**Reply by heygar** [reply]
Just so you know, it's not just Enpass. My Microsoft Authenticator pops up as well asking to save the password. As least I seem to remember it being Microsoft. It may be the Google one. Whichever one I have configured. It's **not** Enpass though.

**Reply by databoy2k** [reply]
Is there a reason for it to be a "textVisiblePassword" input type? Is that necessary for the software?

xDrip is the only software that I've come across as triggering this issue. It's good to see @heygar has the same issue. It's probably something that only certain password managers care about (maybe Enpass is particularly aggressive in trying to detect password fields) but the issue comes back to whether it's necessary for it to be a visible password. Honestly, I mentioned it's low priority because it's just an extra tap to make it go away. I know you're busy with WAY more important and mission-critical bugs. 

I'll see if I'm able to pull a copy of the software, make the change on my end, and if it goes away. As long as there's no technical reason making that inputType critical and if it fixes it, maybe we could evaluate which if any of the fields should have that inputType?

**Reply by Navid200** [reply]
I see 20 different cases in xDrip using this format.  So, the TXID is not the only one.

I suspect this is used so that the keyboard will not give you completion suggestions.

If this is really a problem and needs to be changed, the other 19 instances will need to be dealt with as well.

It doesn't hurt to compile your version and confirm that changing this will address the issue.
If it does, you will need to find an alternative way to stop the keyboard from auto completing.
